docs/source/features/communication_overlap.rst

```diff
@@ -0,0 +1,61 @@
+Communication Overlap
+====================
+
+Data-parallel communication overlap
+-----------------------------------
+
+NeMo supports the overlap of the data-parallel (DP) communications with the computations in LLM training.
+NeMo features Distributed Optimizer that distributes optimizer states and the high precision master parameters across GPUs and this introduces two types of data-parallel communications; reduce-scatter of gradients and all-gather of updated parameters.
+The DP communication is chunked by the granularity of a Transformer layer and overlap each communication chunk with computation.
+This overlap method exposes only one DP communication chunk ensuring efficient large-scale LLM training.
+When training with pipeline-parallelism, the granularity of DP communication becomes the Transformer layers per virtual pipeline stage.
```

Comment:
NeMo includes a Distributed Optimizer that distributes optimizer states and high-precision master parameters across GPUs. This introduces two types of data-parallel communications:

- Reduce-scatter of gradients: This operation splits the gradients across GPUs, allowing efficient communication.

- All-gather of updated parameters: After computation, the updated parameters are gathered from all GPUs.

The data-parallel (DP) communication is chunked based on the granularity of a Transformer layer. These chunks overlap with computation, ensuring efficient large-scale language model (LLM) training. Specifically, only one DP communication chunk is exposed at a time, which helps manage communication overhead.

When training with pipeline parallelism, the granularity of DP communication corresponds to the number of Transformer layers per virtual pipeline stage.